packages/input-container/src/vaadin-input-container.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * @license
+ * Copyright (c) 2021 Vaadin Ltd.
+ * This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+ */
+import { PolymerElement, html } from '@polymer/polymer';
+import { ThemableMixin } from '@vaadin/vaadin-themable-mixin';
+
+export class InputContainer extends ThemableMixin(PolymerElement) {
+  static get is() {
+    return 'vaadin-input-container';
+  }
+
+  static get template() {
+    return html`
+      <style>
+        :host {
+          display: flex;
+          align-items: center;
+          flex: auto;
+        }
+
+        :host([hidden]) {
+          display: none !important;
+        }
+
+        /* Reset the native input styles */
+        ::slotted(input) {
+          -webkit-appearance: none;
+          -moz-appearance: none;
+          outline: none;
+          flex: auto;
+          white-space: nowrap;
+          overflow: hidden;
+          width: 100%;
+          height: 100%;
+          margin: 0;
+          padding: 0;
+          border: 0;
+          border-radius: 0;
+          min-width: 0;
+          font: inherit;
+          font-size: 1em;
+          line-height: normal;
+          color: inherit;
+          background-color: transparent;
+          /* Disable default invalid style in Firefox */
+          box-shadow: none;
+        }
+
+        ::slotted(*) {
+          flex: none;
+        }
+      </style>
+      <slot name="prefix"></slot>
+      <slot name="input"></slot>
```

Comment:
doesn't make much of difference but can we make this the default slot?

Reply:
Good point, removed the `name` from this slot so it will be the default.